src/app/modules/twenty-four-hour-time/twenty-four-hour-time.component.html

```diff
@@ -0,0 +1,168 @@
+    <app-alert dataAutoId="{{ id + '-' + alertVariation }}" [hideClose]="true" variation="{{ alertVariation }}">
+      <div *ngIf="alertMessageList.length > 1; else singleAlert">
+        <div *ngFor="let alertMessage of alertMessageList">
+          <li>{{ alertMessage }}</li>
+        </div>
+      </div>
+      <ng-template #singleAlert>
+        {{ alertMessageList[0] }}
+      </ng-template>
+    </app-alert>
+  </div>  
+</ng-template>
+
+<div aria-live="assertive" class="sr-only">
+  <div *ngIf="screenReaderAnnouncement.length > 0">{{screenReaderAnnouncement}}</div>
+</div>
+
+<div
+  id="time-container-{{id}}"
+  [ngClass]="inputClass + ' ' + 
+    (errorMessage ? 'time-container-error-border' : 'time-container-border') + ' ' +
+    (disabled ? 'disabled-background' : 'enabled-background')"
+  class="time-container
+    ds-l-row
+    ds-u-margin-y--1"
+  >
+  <div class="ds-l-col-10 input-padding">
+    <input
+      class="time-input ds-u-font-size--sm"
+      name="{{ id }}"
+      attr.aria-label="{{ ariaLabel || title }}"
+      attr.data-auto-id="{{ dataAutoId }}"
+      type="text"
+      tabindex="0"
+      onClick="this.setSelectionRange(0, this.value.length)"
+      maxlength="5"
+      [disabled]="disabled"
+      [formControl]="control"
+      [formlyAttributes]="formlyAttributes"
+      (input)="validateInput($event)"
+      (change)="formatAndSetTime($event)"
+      (blur)="blurCallBackFunction($event)"
+      (keyup)="keyupCallBackFunction($event)"
+      (focus)="addFocusStyling()"
+    />
+  </div>
+  <div class="ds-l-col--2 clock-parent">
+    <span
+      role="button"
+      tabindex="0"
+      alt="clock"
+      class="clock-image"
+      aria-label="Activate enter to open the time picker."
+      (click)="openTimePicker($event)"
+      (keyup.enter)="openTimePicker($event)"
+    ></span>
+  </div>
+</div>
+<div class="time-picker
+    ds-u-radius
+    ds-u-border--1
+    ds-u-border--dark"
+  *ngIf="timePickerVisible"
+  id="time-picker-{{id}}"
+  (click)="stopPropagation($event)"
+  (focusout)="focusLost($event)"
+>
+  <div class="time-picker-header">
+    <div tabindex="-1" class="time-picker-label">Select time</div>
+    <span
+      id="{{id}}_close_time_picker"
+      tabindex="0"
+      alt="close"
+      class="close-image"
+      aria-label="Activate enter to close the time picker"
+      (click)="closeTimePicker($event)"
+      (keyup.enter)="closeTimePicker($event)"
+    ></span>
+  </div>
+  <div class="time-picker-body" tabindex="-1">
+    <div class="other-number-row">
+      <div class="other-number-column other-column-top">
+        <div
+          class="bumper-icon top-icon"
+          tabindex="0"
+          aria-label="Activate enter to decrement the hour value"
+          role="button"
+          (click)="stepTime('hours', 'decrement')"
+          (keyup.enter)="stepTime('hours', 'decrement')"
+        >
+          <fa-icon [icon]="faChevronUp"></fa-icon>
+        </div>
```

Comment:
Any opposition to a newline after closing `</div>` tags? I personally do that when it doesn't seem to affect how it gets rendered, as it helps me track element separation and grouping.

That said, I tend to skip the newlines if I have multiple, single-line elements with both open and close tags inline.